simphony_ui/couple_CFDDEM.py

```diff
@@ -22,117 +23,158 @@
 from simliggghts import CUBAExtension
 
 
-def run_calc(output_path, mesh_name):
-    """Executes the full calculation"""
-    openfoam_mode = "internal"
-    mesh_type = "block"
-
-    # Defining the wrapper for OpenFoam
-    if openfoam_mode == "internal":
-        openfoam_wrapper = openfoam_internal.Wrapper()
-        openfoam_cuba_ext = openfoam_internal.CUBAExt
-    elif openfoam_mode == "io":
-        openfoam_wrapper = openfoam_file_io.Wrapper()
-        openfoam_cuba_ext = openfoam_file_io.CUBAExt
-    else:
-        print "Wrong mode_cfd!"
-        sys.exit(1)
-
-    # define the wrapper for LIGGGHTS
-    liggghts_wrapper = liggghts.LiggghtsWrapper(use_internal_interface=True)
-
-    # General settings
-    global_number_iterations = 10
-    global_force_type = "Stokes"
-
-    # OF settings
-    openfoam_num_timesteps = 10
-    openfoam_timestep = 2.0e-4
-    openfoam_viscosity = 1.0e-3
-    openfoam_density = 1000.0
-    openfoam_delta_p = 0.008
-
-    # size of channel in OF
-    openfoam_chansize = [1.0e-1, 1.0e-2, 2.0e-3]
-
-    # number of elements in all channel-directions
-    openfoam_numgrid = [400, 40, 1]
-
-    # **** Liggghts settings *****************
-
-    # The number of DEM steps in each cycle, this is the number of
-    # steps that are run at each call of wrapper.run().
-    liggghts_num_timesteps = 10
-
-    # Time step for MD simulation
-    liggghts_timestep = 1e-6
+class OpenfoamModel:
```

Comment:
This will likely become your HasStrictTraits class which you then proceed to visualise. At this point, it's probably already a good step to just declare so, and have traits for the local storage data. This also solves the problem of the params entry at __init__, which is too anonymous. From outside it's not clear what that dictionary must contain. It also give you a good place to set defaults for all those values.